optaplanner-examples/src/test/java/org/optaplanner/examples/tsp/persistence/LowEntityCountTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.examples.tsp.persistence;
+
+import org.junit.Test;
+import org.optaplanner.core.api.solver.Solver;
+import org.optaplanner.examples.tsp.app.TspApp;
+import org.optaplanner.examples.tsp.domain.TspSolution;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalStateException;
+
+public class LowEntityCountTest {
+
+    public static final long LOW_ENTITY_SECONDS_TERMINATION = 1;
+    public static final String PATH_TO_NEARBY_SELECTION_XML = "org/optaplanner/examples/lowentitycount/nearbySelection.xml";
+    private TspApp app = new TspApp();
+
+    @Test
+    public void zeroEntity() {
+        Solver<TspSolution> tspSolutionSolver = app.createSolver(LOW_ENTITY_SECONDS_TERMINATION);
+        TspSolution problem = app.loadUnsolvedProblemFromResource("oneDomicileOnly.xml");
+        assertThatIllegalStateException().isThrownBy(() -> tspSolutionSolver.solve(problem)).
+                withMessageContaining("annotated member").withMessageContaining("must not return");
+    }
+
+    @Test
+    public void oneEntity() {
+        Solver<TspSolution> tspSolutionSolver = app.createSolver(LOW_ENTITY_SECONDS_TERMINATION);
```

Comment:
Use an attainScore termination with a timeout? Learn from TspPerformanceTest.

Using an entire second is both too long and too short:
- it's too long, because now all our builds on all our machines or all our PR commits for potentially all of KIE are 1 second longer for each time you do this.
- it's too short, because one of jenkins machines will not have enough time with 1 second, and which time someone will make it 10 seconds, which will really increase the build time for the entire KIE team.

Reply:
+1 for using time-independent termination. My preference would be CH only, which doesn't need any termination. OR CH + LS with stepCountLimit=1.

Reply:
+1 for using stepCountLimit instead of a time-based termination.

Reply:
@Fonifan please note that unlike time-spent termination you can't use `stepCountLimit` termination on the solver (global) level. It can only be used on phase level (i.e. inside the `<localSearch>` element because steps only happen inside construction or search phases).